set.mm

```diff
@@ -373630,50 +373916,159 @@ ordered sets (like ` RR ` or ` Prime ` ).  Alternatively, (proper)
+      AWPWDUMDWEUMDWTXBQWQWDUKWEUKWDWEUNULXBWFFFIJZWCWDFWEFIUOXCWCFUPUQURUSUSRU
+      TVAVBVEVCWJWBWKVDZWCWBWJXDWCOWBWJMZWKWCXEAWGGZCSPZBSPZWKWCVFZWJXHWBNSVGZW
+      JXGBNPZXHVHWJWNXFCNPZBNPXKWOWMXLBNWIXFCNWHXFWFWHWGAWHVIVJRTTXLXGBNXJXLXGO
+      VHXFCNSVKVLTVMXGBNSVKVNRWBXHXIQWJBCAVOUJVPVQVRVSVTWA $.
+  $}
+
+  $( Lemma 2 for ~ 2sqreu etc.  (Contributed by AV, 25-Jun-2023.) $)
+  2sqreulem2 $p |- ( ( A e. NN0 /\ B e. NN0 /\ C e. NN0 )
+                     -> ( ( ( A ^ 2 ) + ( B ^ 2 ) ) = ( ( A ^ 2 ) + ( C ^ 2 ) )
+                          -> B = C ) ) $=
+    ( cn0 wcel w3a c2 cexp co caddc wceq nn0cn sqcld 3ad2ant1 3ad2ant2 3ad2ant3
+    cc addcand wi wa nn0sq11 biimpd 3adant1 sylbid ) ADEZBDEZCDEZFZAGHIZBGHIZJI
+    UICGHIZJIKUJUKKZBCKZUHUIUJUKUEUFUIQEUGUEAALMNUFUEUJQEUGUFBBLMOUGUEUKQEUFUGC
+    CLMPRUFUGULUMSUEUFUGTULUMBCUAUBUCUD $.
+
+  $( Lemma 3 for ~ 2sqreu etc.  (Contributed by AV, 25-Jun-2023.) $)
+  2sqreulem3 $p |- ( ( A e. NN0 /\ ( B e. NN0 /\ C e. NN0 ) )
+               -> ( ( ( ph /\ ( ( A ^ 2 ) + ( B ^ 2 ) ) = P )
+                   /\ ( ps /\ ( ( A ^ 2 ) + ( C ^ 2 ) ) = P ) ) -> B = C ) ) $=
+    ( cn0 wcel c2 cexp co caddc wceq wa wi w3a wb eqeq2 eqcoms adantld eqcom ex
+    adantl 2sqreulem2 syl5bi adantr sylbid impd 3expb ) CGHZDGHZEGHZACIJKZDIJKL
+    KZFMZNZBUMEIJKLKZFMZNZNDEMZOUJUKULPZUPUSUTVAUOUSUTOZAVAUOVBVAUONZURUTBVCURU
+    QUNMZUTUOURVDQZVAVEFUNFUNUQRSUCVAVDUTOUOVDUNUQMVAUTUQUNUACDEUDUEUFUGTUBTUHU
+    I $.
+
+  ${
+    $d P b c $.  $d a b c $.  $d ps c $.
+    2sqreulem4.1 $e |- ( ph <-> ( ps /\ ( ( a ^ 2 ) + ( b ^ 2 ) ) = P ) ) $.
+    $( Lemma 4 for ~ 2sqreu et.  (Contributed by AV, 25-Jun-2023.) $)
+    2sqreulem4 $p |- A. a e. NN0 E* b e. NN0 ph $=
+      ( vc cn0 wrmo cv wcel c2 cexp co caddc wceq wa wsbc wral nfcv weq nfsbc1v
+      2sqreulem3 ralrimivva rmobii nfv nfan sbceq1a oveq1 oveq2d eqeq1d anbi12d
+      wi rmo4f bitri sylibr rgen ) AEHIZDHDJZHKZBUSLMNZEJZLMNZONZCPZQZBEGJZRZVA
+      VGLMNZONZCPZQZQEGUAZUMZGHSEHSZURUTVNEGHHBVHUSVBVGCUCUDURVFEHIVOAVFEHFUEVF
+      VLEGHEHTGHTVHVKEBEVGUBVKEUFUGVMBVHVEVKBEVGUHVMVDVJCVMVCVIVAOVBVGLMUIUJUKU
+      LUNUOUPUQ $.
+
+    $( Lemma 2 for ~ 2sqreunn .  (Contributed by AV, 25-Jun-2023.) $)
+    2sqreunnlem2 $p |- A. a e. NN E* b e. NN ph $=
+      ( cn cn0 wss wrmo wral nnssnn0 2sqreulem4 nfcv ssrmof ralimdv mp2 ssralv
+      ) GHIZAEGJZDHKZTDGKLSAEHJZDHKUALABCDEFMSUBTDHAEGHEGNEHNOPQTDGHRQ $.
+  $}
+
+  ${
+    $d P a b c $.
+    2sqreu.1 $e |- ( ph <-> ( a <_ b /\ ( ( a ^ 2 ) + ( b ^ 2 ) ) = P ) ) $.
+    $( There exists a unique decomposition of a prime of the form ` 4 k + 1 `
+       as a sum of squares of two nonnegative integers.  See ~ 2sqnn0 for the
+       existence of such a decomposition.  (Contributed by AV, 4-Jun-2023.)
+       (Revised by AV, 25-Jun-2023.) $)
+    2sqreu $p |- ( ( P e. Prime /\ ( P mod 4 ) = 1 )
+           -> ( E! a e. NN0 E. b e. NN0 ph /\ E! b e. NN0 E. a e. NN0 ph ) ) $=
```

Comment:
perhaps if the hypothesis didn't require `a <_ b` then `E! a e. NN0 E. b e. NN0 ph` would suffice, since addition is commutative so you could do

```
2sqreu
  ( ( P e. Prime /\ ( P mod 4 ) = 1 ) -> E! a e. NN0 E. b e. NN0 ph )
2sqreu
  ( ( P e. Prime /\ ( P mod 4 ) = 1 ) -> E! b e. NN0 E. a e. NN0 ph )
```

Reply:
Yes, the idiom "( E! a e. NN0 E. b e. NN0 ph /\ E! b e. NN0 E. a e. NN0 ph )" represents "there are unique a and b ...", and that is what is shown here.